From 2d5f476f5e7556e6091ed69acfaba3411588b840 Mon Sep 17 00:00:00 2001
From: Anu Engineer <aengineer@apache.org>
Date: Thu, 10 Nov 2016 14:38:53 -0800
Subject: [PATCH 2288/2748] HDFS-8307. Spurious DNS Queries from hdfs shell.
 Contributed by  Andres Perez

(cherry picked from commit 95edf265b3878784a6f239a5d46403a425b534a4)

Conflicts:
hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java

Change-Id: If1066adf8ce2cb5b4a132f6f54b10ff39fdf8286
---
 .../org/apache/hadoop/hdfs/NameNodeProxies.java    |   14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
index d535de0..d0732d7 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/NameNodeProxies.java
@@ -192,8 +192,18 @@ public InetSocketAddress getAddress() {
         dtService = SecurityUtil.buildTokenService(
             NameNode.getAddress(nameNodeUri));
       }
-      return new ProxyAndInfo<T>(proxy, dtService,
-          NameNode.getAddressCheckLogical(conf, nameNodeUri));
+
+      InetSocketAddress nnAddress;
+
+      //We dont need to resolve the address if is it a Nameservice ID
+      if(HAUtil.isLogicalUri(conf, nameNodeUri)) {
+        nnAddress = InetSocketAddress.createUnresolved(
+            nameNodeUri.getHost(), NameNode.DEFAULT_PORT);
+      } else {
+        nnAddress = NameNode.getAddressCheckLogical(conf, nameNodeUri);
+      }
+
+      return new ProxyAndInfo<T>(proxy, dtService, nnAddress);
     }
   }
   
-- 
1.7.9.5

